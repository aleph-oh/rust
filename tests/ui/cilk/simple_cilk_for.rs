#![feature(cilk)]
// Tests a simple cilk_for loop.

// run-pass
// compile-flags: -C panic=abort
// no-prefer-dynamic

use std::sync::atomic::AtomicU64;

fn main() {
    let sum: AtomicU64 = AtomicU64::new(0);
    cilk_for i in 1..10 {
        sum.fetch_add(i, std::sync::atomic::Ordering::Relaxed);
    }
    assert_eq!(sum.into_inner(), 45);
}
