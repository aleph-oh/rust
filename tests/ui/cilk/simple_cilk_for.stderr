note: <unknown file>:0:0 loop-stripmine (other): Parallel loop does not appear profitable to parallelize.

